cyipopt/scipy_interface.py

```diff
@@ -323,6 +323,33 @@ def convert_to_bytes(options):
                 pass
 
 
+def _wrap_fun(fun, kwargs):
+    if callable(fun) and kwargs:
+        def new_fun(x, *args):
+            return fun(x, *args, **kwargs)
+    else:
+        new_fun = fun
+    return new_fun
+
+def _wrap_funs(fun, jac, hess, hessp, constraints, kwargs):
+    wrapped_fun = _wrap_fun(fun, kwargs)
+    wrapped_jac = _wrap_fun(jac, kwargs)
+    wrapped_hess = _wrap_fun(hess, kwargs)
+    wrapped_hessp = _wrap_fun(hessp, kwargs)
+    if isinstance(constraints, dict):
+        constraints = (constraints,)
+    wrapped_constraints = []
+    for constraint in constraints:
+        constraint = constraint.copy()
+        ckwargs = constraint.pop('kwargs', {})
+        constraint['fun'] = _wrap_fun(constraint.get('fun', None), ckwargs)
+        constraint['jac'] = _wrap_fun(constraint.get('jac', None), ckwargs)
+        wrapped_constraints.append(constraint)
+    return (wrapped_fun, wrapped_jac, wrapped_hess, wrapped_hessp,
+            wrapped_constraints)
```

Comment:
`minimize` does not natively support `kwargs` like `minimize_ipopt`, so we have to do some wrapping. This is why the tests heavily emphasize the callable arguments (and I didn't test `bounds`, `options`, etc.).

There is one other callable, `callback`, but that doesn't need to be wrapped because it does not accept additional arguments.

Reply:
The kwargs haven't ever worked before, I added them in the recent PR #197. If i makes more sense to disable the support for kwargs, we can.

Reply:
Well, they work now, so no harm leaving them unless you want to deprecate them and remove the parameter altogether.

Reply:
They don't need to be deprecated because we haven't made a new release yet. If the goal is to match the scipy api, then maybe we should stick to that. Otherwise, I'm open fro whatever people need.

Reply:
I know it didn't work before, but it looks like it was already present as a positional/keyword argument in a previous release (e.g. [1.0.3](https://cyipopt.readthedocs.io/en/v1.0.3/reference.html)). For SciPy, that would mean it needs to be deprecated before it can be removed because working user code may pass it as an argument (even though it does nothing). But if you are comfortable with the (probably rare) immediate disruption of removing it (e.g. now be an error if it is passed), I understand.

I'd rather not weigh in on that decision, though. (If I had my way, `minimize` would have neither `args` nor `kwargs`, `jac` would be `grad` - see https://github.com/scipy/scipy/pull/18249, etc...)

Reply:
@moorepants since it was already present as an argument in previous releases, do you agree that it would need to be deprecated? In light of that, would you like to just keep it?

Reply:
I'm fine keeping it, but we should document how we deviate from scipy's minimize function.

Reply:
I added a bit at the top of the documentation.

Please don't keep `kwargs` on my account, though. As I mentioned, I am almost always against accepting `args` and `kwargs` just to be passed through to functions. I would prefer that standard Python features be used to accomplish the same thing (e.g. `def`, `lambda`, or `functools.partial` to wrap the callable). I've supported them here because whether they worked or not, they have been part of the signature for at least a few releases.